### PR TITLE
curtail: init at 1.3.0

### PR DESCRIPTION
**Pull request by schuelermine**
###### Description of changes

Adds Curtail, a simple GTK image compressor

This is a third pull request about this. Sorry, I mis-rebased before and the whole thing got bollocksed up. Please see #173520 for previous discussion.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/curtail/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    gettext
+  ];
+
+  # This is needed to let gobject-introspection’s setup hook run
+  # Disabling this breaks icon themes (at least)
+  strictDeps = false;
+
+  # Tells the Python builder to use normal building methods instead of using a setup.py
+  format = "other";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/curtail/default.nix:54]:
```suggestion
```

File: pkgs/applications/graphics/curtail/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  propagatedBuildInputs = [
+    appstream-glib
+    python3.pkgs.pygobject3
+    gobject-introspection
+    gettext
+  ];
+
+  # This is needed to let gobject-introspection’s setup hook run
+  # Disabling this breaks icon themes (at least)
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/curtail/default.nix:49]:
```suggestion
  # Currently still required for the gobject-introspection setup hook
```

File: pkgs/applications/graphics/curtail/default.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  strictDeps = false;
+
+  # Tells the Python builder to use normal building methods instead of using a setup.py
+  format = "other";
+
+  preInstall = ''
+    patchShebangs --build /build/source/build-aux/meson/postinstall.py
```
[the remark this answers is not in the document]
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix:56]: This breaks the build. `patchShebangs` fails silently and the script cannot run.

File: pkgs/applications/graphics/curtail/default.nix (hunk 1 above)
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix]: Done

Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix]: Done

File: pkgs/applications/graphics/curtail/default.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+, gobject-introspection
+, jpegoptim
+, libwebp
+, optipng
+, pngquant
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "curtail";
+  version = "1.3.0";
```
[the remark this answers is not in the document]
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix:21]: Done

File: pkgs/applications/graphics/curtail/default.nix (hunk 2 above)
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix]: Done

File: pkgs/applications/graphics/curtail/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix]: I’ve removed `--build`, but removing the /build/source/ fails.

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/graphics/curtail/default.nix]: We shouldn't use an absolute path here. Can you figure out a relative path and use that?

[the remark this answers is not in the document]
Comment (schuelermine) [reply pkgs/applications/graphics/curtail/default.nix]: OK, I figured out the relative path.
But why shouldn’t we use the relative path?

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/graphics/curtail/default.nix]: I am not sure if it is consistent across linux/darwin and it is general bad practise.

**Pull request merged**